### PR TITLE
[OpenMP] Support setting POSIX thread name on *BSD's and Solaris

### PR DESCRIPTION
**Pull request by brad0**

### CODE REVIEW

File: openmp/runtime/src/z_Linux_util.cpp [hunk 1]
```diff
@@ -878,10 +881,19 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
...
   }
 
-#ifdef LIBOMP_HAVE_LINUX_PTHREAD_SETNAME
+#if defined(LIBOMP_HAVE_PTHREAD_SETNAME_NP) ||                                 \
+    defined(LIBOMP_HAVE_PTHREAD_SET_NAME_NP)
   // Rename worker threads for improved debuggability
   if (!KMP_UBER_GTID(gtid)) {
+#if defined(__linux__)
     pthread_setname_np(handle, "openmp_worker");
+#elif defined(__FreeBSD__) || defined(__OpenBSD__)
```
Comment (yamt) [inline openmp/runtime/src/z_Linux_util.cpp:890]: a question; why in some places do you check `KMP_OS_OPENBSD` and in other places `__OpenBSD__`?

Comment (brad0) [reply openmp/runtime/src/z_Linux_util.cpp]: That's a good point. I have fixed that.

File: openmp/runtime/src/z_Linux_util.cpp [hunk 2]
```diff
@@ -878,10 +881,19 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
...
   if (!KMP_UBER_GTID(gtid)) {
+#if KMP_OS_LINUX
     pthread_setname_np(handle, "openmp_worker");
+#elif KMP_OS_FREEBSD || KMP_OS_OPENBSD || KMP_OS_DRAGONFLY
+    pthread_set_name_np(handle, "openmp_worker");
+#elif KMP_OS_NETBSD
+    pthread_setname_np(handle, "%s", const_cast<char *>("openmp_worker"));
+#elif KMP_OS_DARWIN
+    pthread_setname_np("openmp_worker");
+#endif
```
Comment (shiltian) [inline openmp/runtime/src/z_Linux_util.cpp:896]: I'd do something like:

```
#if defined(LIBOMP_HAVE_PTHREAD_SET_NAME_NP)
pthread_set_name_np(handle, "openmp_worker");
#elif defined(LIBOMP_HAVE_PTHREAD_SETNAME_NP)
#if ...
#elif ...
...
#endif
```

File: openmp/runtime/src/z_Linux_util.cpp [hunk 3]
```diff
@@ -878,10 +881,19 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
...
   // Rename worker threads for improved debuggability
   if (!KMP_UBER_GTID(gtid)) {
+#if KMP_OS_LINUX
     pthread_setname_np(handle, "openmp_worker");
+#elif KMP_OS_FREEBSD || KMP_OS_OPENBSD || KMP_OS_DRAGONFLY
+    pthread_set_name_np(handle, "openmp_worker");
+#elif KMP_OS_NETBSD
+    pthread_setname_np(handle, "%s", const_cast<char *>("openmp_worker"));
+#elif KMP_OS_DARWIN
+    pthread_setname_np("openmp_worker");
```
Comment (MatzeB) [inline openmp/runtime/src/z_Linux_util.cpp:895]: This sets the name of the current thread though? But what we want is to change the name of the newly created thread pointed to by `handle`?

Comment (brad0) [reply openmp/runtime/src/z_Linux_util.cpp]: Good point. I took out the macOS handling.

**Review by shiltian — APPROVED** [review]

**Pull request merged**